spec/support/matchers.rb

```diff
@@ -20,7 +23,7 @@
 
 RSpec::Matchers.define :be_cstring do |string|
   match do |actual|
-    actual == "#{string.force_encoding(BSON::BINARY)}\0"
+    actual == "#{string.dup.force_encoding(BSON::BINARY)}\0"
```

Comment:
Should this be in a separate commit to differentiate it from this PR?

Reply:
I do not understand what you mean. This PR makes string literals produce frozen strings. Sometimes the strings thus produced are modified by the code, in which case they need to be unfrozen. This is such an occurrence.

Reply:
Yeah this was my mistake. I should have done a bit more research here.